File: Ruby/Summation_of_Primes.rb
# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
#
# Find the sum of all the primes below two million.

def sieve(n)
  list = Array.new(n,true)
  crossLimit = Math.sqrt(n).floor
  output = []

  (2..crossLimit).each do |i|
    if list[i]
      j = i * i
        while j < n
          list[j] = false
          j += i
        end
    end
  end

  (2..n).each do |i|
    if list[i]
      output << i
    end
  end


  output
end

def sum_prime(num)
  sieve(num).inject(&:+)
end

File: Javascript/Summation_of_Primes.js
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
//
// Find the sum of all the primes below two million.

function sieve(n) {
    var list = []
    var crossLimit = Math.floor(Math.sqrt(n))
    var output = []

    for (var i = 0; i < n; i++) {
        list.push(true)
    }

    for (var i = 2; i <= crossLimit; i++) {
        if (list[i]) {
            for (var j = i * i; j < n; j += i) {
                list[j] = false
            }
        }
    }

    for (var i = 2; i < n; i++) {
        if (list[i]) {
            output.push(i)
        }
    }

    return output
}

function sum_prime(num) {
  return sieve(num).reduce(function(a,b) {
    return a + b;
  });
}
